test/test_ops.py

```diff
@@ -463,6 +439,115 @@ def _fn(*sample_args, **sample_kwargs):
                 graph = traced.graph_for(*inp)
                 FileCheck().check(op_name).check_not(variant_name).run(graph)
 
+    # Validates ops implement the correct out= behavior
+    # See https://github.com/pytorch/pytorch/wiki/Developer-FAQ#how-does-out-work-in-pytorch
+    #   for a description of the correct behavior
+    @ops(op_db, dtypes=OpDTypes.supported, allowed_dtypes=(torch.float,))
+    def test_out(self, device, dtype, op):
+        # TODO: verify the op doesn't suppor the out= kwarg
+        if not op.supports_out:
+            self.skipTest("Skipped! Op doesn't support out= kwarg.")
+
+        # NOTE: only tests on first sample
+        samples = op.sample_inputs(device, dtype)
+        sample = samples[0]
+
+        # calls it normally to get the expected result
+        expected = op(*sample.input, *sample.args, **sample.kwargs)
+        op_out = partial(op, *sample.input, *sample.args, **sample.kwargs)
+
+        # Short-circuits if output is not a single tensor
+        # TODO: extend this to handle multiple tensors
+        if not isinstance(expected, torch.Tensor):
+            self.skipTest("Skipped! Only supports single tensor outputs for now.")
+
+        # Case 0: out= with the correct shape, dtype, and device
+        #   but NaN values
+        #   Expected behavior: out= values have no effect on the computation.
+        out = torch.full_like(expected, float('nan'))
+        op_out(out=out)
+        self.assertEqual(expected, out)
+
+        # Case 1: out= with the correct shape, dtype, and device,
+        #   but noncontiguous.
+        #   Expected behavior: strides are respected.
+        # TODO: (implement) and switch to make_tensor_like()
+        out = make_tensor(expected.shape,
+                          dtype=expected.dtype,
+                          device=expected.device,
+                          discontiguous=True)
+        original_strides = out.stride()
+        op_out(out=out)
+        self.assertEqual(expected, out)
+        self.assertEqual(original_strides, out.stride())
+
+        # Case 2: out= with the correct dtype and device, but the wrong shape
+        #   Expected behavior: resize with a warning.
+        wrong_shape = list(expected.shape)
```

Comment:
TODO: need to detect scalar tensor case separately (and pass 2 element 1D tensor, for example, in that case)